### PR TITLE
feat: Convenience factories for using with Akka Projection gRPC

### PR DESCRIPTION
**Pull request by johanandren**
References #351

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -71,12 +73,14 @@ object Dependencies {
...
     r2dbcPostgres,
     akkaProjectionCore,
+    akkaProjectionGrpc,
     TestDeps.akkaProjectionEventSourced,
     TestDeps.akkaProjectionDurableState,
     TestDeps.akkaStreamTestkit,
     TestDeps.akkaTestkit,
     TestDeps.akkaProjectionTestKit,
     TestDeps.akkaJackson,
+    TestDeps.akkaDiscovery,
```
Comment (patriknw) [inline project/Dependencies.scala:83]: why is this needed?

**Review by patriknw — COMMENTED** [review]
nice

File: project/Dependencies.scala (hunk 1 above)
Comment (johanandren) [reply project/Dependencies.scala:83]: Not sure, tests started failing with version mismatch between discovery and the rest of the Akka deps, but Test/dependencyTree doesn't show it 🤔 

Does Akka gRPC pull it in for client service discovery perhaps? But no-show in dep tree is a mystery to me.

File: project/Dependencies.scala [hunk 2]
```diff
@@ -71,12 +73,14 @@ object Dependencies {
     r2dbcPool,
     r2dbcPostgres,
     akkaProjectionCore,
+    akkaProjectionGrpc,
```
[the remark this answers is not in the document]
Comment (johanandren) [reply project/Dependencies.scala:76]: I did mark it optional up in the value definition, also put the factories in separate classes from R2dbcProjection so it isn't needed on classpath to use the existing projection factories.

File: project/Dependencies.scala (hunk 1 above)
Comment (patriknw) [reply project/Dependencies.scala:83]: ok, I don't know, maybe something that resolves differently for milestones (I think I have seen such problems before with milestones)

**Review by patriknw — APPROVED** [review]
LGTM

File: project/Dependencies.scala [hunk 3]
```diff
@@ -20,6 +20,7 @@ object Dependencies {
     val akkaPersistenceQuery = "com.typesafe.akka" %% "akka-persistence-query" % AkkaVersion
 
     val akkaProjectionCore = "com.lightbend.akka" %% "akka-projection-core" % AkkaProjectionVersion
+    val akkaProjectionGrpc = "com.lightbend.akka" %% "akka-projection-grpc" % AkkaProjectionVersion % Optional
```
[the remark this answers is not in the document]
Comment (johanandren) [reply project/Dependencies.scala:23]: Let's switch to provided
